snakemake/logging.py

```
@@ -305,12 +243,504 @@
+        return f"[{time.asctime()}]"
+
+    def format_percentage(self, done, total):
+        """Helper method to format percentage."""
+        if done == total:
+            return "100%"
+        if done == 0:
+            return "0%"
+        precision = 0
+        fraction = done / total
+        fmt_precision = "{{:.{}%}}".format
+        fmt = lambda fraction: fmt_precision(precision).format(fraction)
```

Ruff

Do not assign a `lambda` expression, use a `def`

Rewrite `fmt` as a `def`

(E731)